src/prefect/engine.py

```diff
@@ -131,6 +134,7 @@ def enter_flow_run_engine_from_flow_call(
         create_and_begin_subflow_run if is_subflow_run else create_then_begin_flow_run,
         flow=flow,
         parameters=parameters,
+        wait_for=wait_for if is_subflow_run else None,
```

Comment:
Is this just a safety guard?

Reply:
I'm not sure we actually require it to be a subflow, it'd be weird to pass things and have them be ignored. Maybe we should throw an exception a few lines up if it's not a subflow run but it is not none?

Reply:
I have a preference for an exception instead of silently ignoring it. I'm not really sure what users would be passing in, but I think it'll be clearer to discourage it with an error.